### ISSUE TITLE
Resume bucket value if forecast rainfall is not met

### ISSUE THREAD

**Issue by jlrosssc** [jeroenterheerdt/HAsmartirrigation#423]
I've had a few instances recently where rain was forecast and the integration appropriately took that into account, however the rain never came and it should have watered.  It would be great if the integration compared the anticipated, forecasted rainfall vs what my rain gauge actually reads and adjust accordingly.  It seems as if once the forecast rainfall is calculated, it's never undone or modified that amount is not met.

**Comment by jeroenterheerdt** [comment]
how would this work? a forecast is a forecast, so at the time of calculation the only thing we can do is take into account tomorrow's (or the days after's) forecast(s) and adjust tomorrow mornings irrigation accordingly. I fail to see how we would achieve what you describe. As an example:
let's say on saturday night the irrigation amount is calculated for sunday morning. At that time there is rain forecasted for sunday. So we decide not to irrigate on Sunday morning. At that time the rain gauge reads 0, but we don't know if rain is still coming on Sunday later that day. It doesn't actually rain on sunday so in retrospect you would say we should have irrigated but there is no way of us knowing on Sunday morning. The only thing that happens here is that we skipped Sunday but will most likely irrigate on Monday morning unless again the forecast is way off.

Makes sense? Feel free to provide other options.

**Comment by jlrosssc** [comment]
I see your point about the rain possibly falling later in the day.  I'm not a coder by any means, but what I was envisioning to make this as accurate as possible, is a placeholder for the anticipated rain impact.  For example, if the forecast anticipates .5" rain and the ET calculation takes that into account, place that amount in a placeholder.  At the end Sunday in your example, you would "true up" and add anything left in that placeholder.  If no rain was detected, all of it would be added back.  .25", would mean half of the placeholder would be added back.

**Comment by jeroenterheerdt** [comment]
but ET is not equal to rainfall ,so this is not going to work.

**Comment by acarlo79** [comment]
What is the current logic? If we get rain forecast, are you just using the amount of rain to calculate the new bucket value? I think it would make more sense to never use forecast value but rather setting a flag to be used in our outomation if there is rain forecast in the next 24hrs.
In this way, your bucket calculation will be always accurate and you avoid to water if there is rain....if the rain won't arrive, the new bucket will be calculated correctly again and you will water again the following day.

**Comment by jlrosssc** [comment]
That is essentially the concept that I was trying to articulate as well. Sent from my iPhoneOn Jul 23, 2024, at 1:14 PM, acarlo79 ***@***.***> wrote:﻿
What is the current logic? If we get rain forecast, are you just using the amount of rain to calculate the new bucket value? I think it would make more sense to never use forecast value but rather setting a flag to be used in our outomation if there is rain forecast in the next 24hrs.
In this way, your bucket calculation will be always accurate and you avoid to water if there is rain....if the rain won't arrive, the new bucket will be calculated correctly again and you will water again the following day.

**Comment by jeroenterheerdt** [comment]
> What is the current logic? If we get rain forecast, are you just using the amount of rain to calculate the new bucket value? I think it would make more sense to never use forecast value but rather setting a flag to be used in our outomation if there is rain forecast in the next 24hrs. In this way, your bucket calculation will be always accurate and you avoid to water if there is rain....if the rain won't arrive, the new bucket will be calculated correctly again and you will water again the following day.

correct, the amount of forecasted rain is taken into account to calculate the new bucket value.

**Comment by acarlo79** [comment]
I would maybe change this part since forecast is just a "guess" and could lead to a wrong bucket beeing calculated.

I would just expose a value to tell (rain is forecasted in the next 24hrs) so you could skip tomorrow watering and then calculate the new bucket correctly (with or without the rain fall of that day).

I think would be possible already now if you take a forecast rain sensor and set the forecast days to 0.

**Comment by jeroenterheerdt** [comment]
Introduced in 2025.7.2
